corehq/apps/products/forms.py

```diff
@@ -83,7 +83,7 @@ def clean_code(self):
         sms_codes = all_sms_codes(self.product.domain)
         if code in sms_codes:
             conflict = sms_codes[code]
-            if conflict[0] == 'product' and conflict[1].id == self.product.id:
+            if conflict[0] == 'product' and conflict[1].product_id == self.product._id:
```

Comment:
The conflict's product is a `SQLProduct`: https://github.com/dimagi/commcare-hq/blob/3abc0168e9892d154c5a61a484d28a0e9c449ba1/corehq/apps/commtrack/util.py#L36

`self.product` is a couch `Product`: https://github.com/dimagi/commcare-hq/blob/37313634de2b5b6818d5e32ecd66a0a1c7e76c02/corehq/apps/products/views.py#L223